src/devtool/debug_state.zig

```diff
@@ -301,14 +317,29 @@ pub fn freeEvmStateJson(allocator: std.mem.Allocator, state: EvmStateJson) void
     allocator.free(state.logs);
 
     allocator.free(state.returnData);
+
+    // Free blocks
+    for (state.blocks) |blk| {
+        allocator.free(blk.pcs);
+        for (blk.opcodes) |s| allocator.free(s);
+        allocator.free(blk.opcodes);
+        for (blk.hex) |s| allocator.free(s);
+        allocator.free(blk.hex);
+        for (blk.data) |s| allocator.free(s);
+        allocator.free(blk.data);
+        allocator.free(blk.instIndices);
+        allocator.free(blk.instMappedPcs);
+    }
+    allocator.free(state.blocks);
```

Comment:
In future we can make this simpler by just using an arena allocator. An arena allocator allocates a bunch of times and tehn frees everything at once